components/script/dom/offscreencanvas.rs

```diff
@@ -126,8 +128,8 @@ impl OffscreenCanvas {
         Some(context)
     }
 
-    pub(crate) fn placeholder(&self) -> Option<&HTMLCanvasElement> {
-        self.placeholder.as_deref()
+    pub(crate) fn placeholder(&self) -> Option<&WeakRef<HTMLCanvasElement>> {
+        self.placeholder.as_ref()
     }
 }
```

Comment:
Weakref somehow acts as an option, if the type is GCed .root() will return none and Some otherwise. So here we can do something like this:
```suggestion
    pub(crate) fn placeholder(&self) -> Option<DomRoot<HTMLCanvasElement>> {
        self.placeholder.and_then(|placeholder| placeholder.root())
    }
}
```
I think that should solve all your problems.

Reply:
So I tried this thing, but had to clone canvas in one place and also added  many `as_deref()`'s. Not sure if it is ok, but this is the best I can do now.